pkg/services/thumbs/service.go

```diff
@@ -209,13 +213,68 @@ func (hs *thumbService) UpdateThumbnailState(c *models.ReqContext) {
 	c.JSON(http.StatusOK, map[string]string{"success": "true"})
 }
 
+func getDatasourceUIDs(resp *backend.DataResponse) ([]string, error) {
+	if resp == nil {
+		return nil, errors.New("nil response")
+	}
+
+	if resp.Error != nil {
+		return nil, resp.Error
+	}
+
+	if len(resp.Frames) == 0 {
+		return nil, errors.New("empty response")
+	}
+
+	frame := resp.Frames[0]
+	field, idx := frame.FieldByName("ds_uid")
+
+	if field.Len() == 0 || idx == -1 {
+		return nil, errors.New("no ds_uid field")
+	}
+
+	rawValue, ok := field.At(0).(*json.RawMessage)
+	if !ok || rawValue == nil {
+		return nil, errors.New("invalid value in ds_uid field")
+	}
+
+	jsonValue, err := rawValue.MarshalJSON()
+	if err != nil {
+		return nil, err
+	}
+
+	var uids []string
+	err = json.Unmarshal(jsonValue, &uids)
+	if err != nil {
+		return nil, err
+	}
+
+	return uids, nil
+}
+
 func (hs *thumbService) GetImage(c *models.ReqContext) {
 	req := hs.parseImageReq(c, false)
 	if req == nil {
 		return // already returned value
 	}
 
-	res, err := hs.thumbnailRepo.getThumbnail(c.Req.Context(), models.DashboardThumbnailMeta{
+	ctx := c.Req.Context()
+
+	dashQueryResponse := hs.searchService.DoDashboardQuery(ctx, &backend.User{
```

Comment:
rather than implementing the ds permissions check here... perhaps we can add that to the search service?  ie it would only return dashboards that you have permissions to read + read all data.

Then we could avoid the awkward decoding here